pandas/tests/test_window.py

```diff
@@ -275,6 +275,17 @@ def test_how_compat(self):
                             getattr(s, t)(freq='D', **kwargs), op)(how=how)
                         assert_series_equal(result, expected)
 
+    # GH #12373 : rolling functions error on float32 data
+    def test_ensure_type_float64(self):
+        expected = pd.Series([np.nan, np.nan, 2, 3, 4])
+
```

Comment:
right, sorry, I actually made work. Ok, then need some tests for this explictiy (which are already there IIRC), so pls move this test next to those and can check the dtypes for different item sizes then. we have lots of tests :)